### ISSUE TITLE
health checks for Applications with multiple sources can probably work now

### ISSUE THREAD

**Issue by krancour** [akuity/kargo#1399]
### Discussed in https://github.com/akuity/kargo/discussions/1398

<div type='discussions-op-text'>

<sup>Originally posted by **daftping** January 18, 2024</sup>
Seems like the health check for Applications with multiple sources is not supported right now.
https://github.com/akuity/kargo/blob/8b6f58a4ddfc81e004bb782de69a45a575d29a44/internal/controller/stages/health.go#L98-L100
Does anybody know what Argo CD issue this TODO is referring to?

```yaml
  health:
    status: Unknown
    issues:
      - bugs in Argo CD currently prevent a comprehensive assessment of the
        health of multi-source Application "test-app-helm" in
        namespace "argo-cd"
```</div>

**Comment by krancour** [comment]
Refactoring health checks is a long outstanding todo anyway.

I have long been intending to make health checks a function of promotion mechanisms. i.e. A promotion mechanism should describe how to promote _and_ how to assert success.

**Comment by krancour** [comment]
Closed by #2552
